File: src/main/java/com/svantle/score/listeners/PlayerDataListener.java
package com.svantle.score.listeners;

import com.svantle.score.SCore;
import com.svantle.score.utils.MessagesHandler;

import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.io.IOException;

public class PlayerDataListener implements Listener {
    private final SCore plugin;
    private final MessagesHandler messagesHandler;

    public PlayerDataListener(SCore plugin) {
        this.plugin = plugin;
        this.messagesHandler = plugin.getMessagesHandler();
    }

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();
        File dataFolder = plugin.getDataFolder();
        File playerFile = new File(dataFolder, "playerdata.yml");
        FileConfiguration playerData = YamlConfiguration.loadConfiguration(playerFile);

        boolean wasFlying = playerData.getBoolean(player.getUniqueId().toString() + ".flight", false);

        player.setAllowFlight(wasFlying);
        if (wasFlying) {
            player.setFlying(true);
            player.sendMessage(messagesHandler.prefixed(messagesHandler.get("flight_enabled")));
        }
    }

    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        Player player = event.getPlayer();
        File dataFolder = plugin.getDataFolder();
        File playerFile = new File(dataFolder, "playerdata.yml");
        FileConfiguration playerData = YamlConfiguration.loadConfiguration(playerFile);

        playerData.set(player.getUniqueId().toString() + ".flight", player.getAllowFlight());

        try {
            playerData.save(playerFile);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/com/svantle/score/listeners/PlayerDataListener.kt
package com.svantle.score.listeners

import com.svantle.score.SCore
import com.svantle.score.constants.FILE_PLAYER_DATA
import com.svantle.score.constants.MSG_ENABLED
import com.svantle.score.constants.MSG_FLIGHT_UPDATED
import com.svantle.score.utils.MessageHelper
import com.svantle.score.utils.sendTo
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.configuration.file.YamlConfiguration
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent
import java.io.File
import java.io.IOException

class PlayerDataListener(private val core: SCore): Listener {

	init {
		core.server.pluginManager.registerEvents(this, core)
	}

	@EventHandler
	fun onPlayerJoin(event: PlayerJoinEvent) {
		val player = event.player
		val dataFolder: File = core.dataFolder
		val playerFile = File(dataFolder, FILE_PLAYER_DATA)
		val playerData: FileConfiguration = YamlConfiguration.loadConfiguration(playerFile)
		val wasFlying = playerData.getBoolean(player.uniqueId.toString() + ".flight", false)

		player.allowFlight = wasFlying

		if(wasFlying) {
			player.isFlying = true

			val enabled = core.messages.getMessageString(MSG_ENABLED)
			MessageHelper.phrase(MSG_FLIGHT_UPDATED, enabled).sendTo(player)
		}
	}

	@EventHandler
	fun onPlayerQuit(event: PlayerQuitEvent) {
		val player = event.player
		val dataFolder: File = core.dataFolder
		val playerFile = File(dataFolder, FILE_PLAYER_DATA)
		val playerData: FileConfiguration = YamlConfiguration.loadConfiguration(playerFile)

		playerData[player.uniqueId.toString() + ".flight"] = player.allowFlight

		try {
			playerData.save(playerFile)
		} catch(e: IOException) {
			e.printStackTrace()
		}
	}
}